bundles/org.openhab.binding.openwebnet/src/main/java/org/openhab/binding/openwebnet/handler/OpenWebNetLightingHandler.java

```diff
@@ -0,0 +1,376 @@
+        if (newWhatInt != latestBrightnessWhat) {
+            if (newWhatInt >= 0 && newWhatInt <= 10) {
+                newWhat = Lighting.WHAT.fromValue(newWhatInt);
+                if (newWhat.equals(Lighting.WHAT.ON)) {
+                    // change it to WHAT.DIMMER_20 (dimming to 10% is not allowed in OWN)
+                    newWhat = Lighting.WHAT.DIMMER_20;
+                }
+                // save current brightness level before sending bri=0 command to device
+                if (newWhatInt == 0) {
+                    latestBrightnessWhatBeforeOff = latestBrightnessWhat;
+                }
+                lastBrightnessChangeSentTS = System.currentTimeMillis();
+                Where w = deviceWhere;
+                if (w != null) {
+                    try {
+                        send(Lighting.requestDimTo(w.value(), newWhat));
+                    } catch (OWNException e) {
+                        logger.warn("Exception while sending dimLightTo for command {}: {}", command, e.getMessage());
+                    }
+                }
+                if (!(command instanceof PercentType)) {
+                    updateState(channel, new PercentType(Lighting.levelToPercent(newWhatInt)));
+                }
+                latestBrightnessWhat = newWhatInt;
+            } else {
+                logger.debug("-DIM- do nothing");
+            }
+        } else {
+            logger.debug("-DIM- do nothing");
+        }
+        logger.debug("-DIM- latestBriWhat={} latestBriBeforeOff={} briLevelRequested={}", latestBrightnessWhat,
+                latestBrightnessWhatBeforeOff, brightnessLevelRequested);
+    }
+
+    @Override
+    protected String ownIdPrefix() {
+        return Who.LIGHTING.value().toString();
+    }
+
+    @Override
+    protected void handleMessage(BaseOpenMessage msg) {
+        super.handleMessage(msg);
+        updateLightState((Lighting) msg);
+    }
+
+    /**
+     * Updates light state based on a OWN Lighting message received
+     *
+     * @param msg the Lighting message received
+     */
+    private void updateLightState(Lighting msg) {
+        logger.debug("updateLightState() for thing: {}", thing.getUID());
+        ThingTypeUID thingType = thing.getThingTypeUID();
+        if (THING_TYPE_ZB_DIMMER.equals(thingType) || THING_TYPE_BUS_DIMMER.equals(thingType)) {
+            updateLightBrightnessState(msg);
+        } else {
+            updateLightOnOffState(msg);
+        }
+    }
+
+    /**
+     * Updates on/off state based on a OWN Lighting message received
+     *
+     * @param msg the Lighting message received
+     */
+    private void updateLightOnOffState(Lighting msg) {
+        String channelID;
+        OpenWebNetBridgeHandler brH = bridgeHandler;
+        if (brH != null && brH.isBusGateway()) {
+            channelID = CHANNEL_SWITCH;
+        } else {
+            if (WhereZigBee.UNIT_02.equals(((WhereZigBee) (msg.getWhere())).getUnit())) {
```

Comment:
There is a check if gateway is BUS or not; in case gw is ZigBee, msgs received from openwebnet protocol must be addressed using WhereZigBee format otherwise the parser in the openwebnet4j lib would fail parsing the msg and drop it. Anyway I made the code more readable